plugins/catalog-react/src/hooks/useEntityOwnership.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CatalogApi } from '@backstage/catalog-client';
+import {
+  Entity,
+  EntityName,
+  parseEntityRef,
+  RELATION_MEMBER_OF,
+  RELATION_OWNED_BY,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import {
+  IdentityApi,
+  identityApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import jwtDecoder from 'jwt-decode';
+import { useMemo } from 'react';
+import { useAsync } from 'react-use';
+import { catalogApiRef } from '../api';
+import { getEntityRelations } from '../utils/getEntityRelations';
+
+// Takes a user ID from the identity, which can be on basically any form, and
+// returns an entity ref. E.g. if the input is "foo", it returns
+// "user:default/foo" to make sure it's a full ref.
+function extendUserId(id: string): string {
+  try {
+    const ref = parseEntityRef(id, {
+      defaultKind: 'User',
+      defaultNamespace: 'default',
+    });
+    return stringifyEntityRef(ref);
+  } catch {
+    return id;
+  }
+}
+
+// Takes the relevant parts of the Backstage identity, and translates them into
+// a list of entity refs on string form that represent the user's ownership
+// connections.
+export async function loadIdentityOwnerRefs(
+  identityApi: IdentityApi,
+): Promise<string[]> {
+  const id = identityApi.getUserId();
+  const token = await identityApi.getIdToken();
+  const result: string[] = [];
+
+  if (id) {
+    result.push(extendUserId(id));
+  }
+
+  if (token) {
+    try {
+      const decoded = jwtDecoder(token) as any;
+      if (decoded?.ent) {
+        [decoded.ent]
+          .flat()
+          .filter(x => typeof x === 'string')
+          .map(x => x.toLocaleLowerCase('en-US'))
+          .forEach(x => result.push(x));
+      }
+    } catch {
+      // ignore
+    }
+  }
+
+  return result;
+}
+
+// Takes the relevant parts of the User entity corresponding to the Backstage
+// identity, and translates them into a list of entity refs on string form that
+// represent the user's ownership connections.
+export async function loadCatalogOwnerRefs(
+  catalogApi: CatalogApi,
+  identityOwnerRefs: string[],
+): Promise<string[]> {
+  const result = new Array<string>();
+
+  const primaryUserRef = identityOwnerRefs.find(ref => ref.startsWith('user:'));
```

Comment:
Is is possible that the `ent` claim includes users? Or is it explicitly documented that the list should then include all group relations since they are not resolved here? I assume that this `loadCatalogOwnerRefs` is mainly a backward compatibility with the old approach.

Reply:
Mostly compatibility yeah. Not sure if long term we'll keep this method too, but it's probably worth trying to abandon it and leverage `ent` fully. It has such great benefits compared to the catalog method.

The claim should contain _all_ of your alternate identities and groups that are potential owners of things, so yep. It could for example contain a `user:github/freben` to list my github identity along with my company-internal one.